### DISCUSSION TITLE
How to add that language button on my Repo?

### DISCUSSION THREAD

**Discussion by rkhattab** [community/community#121733] [New to GitHub] [answered]
Hey Everyone, 

I have done projects in the past, but it seems like Github UI has changed a lot since then. I can't seem to figure out how to add that programming language label so that it appears in the details on my repo homepage. You can see in the attachment that I have the symbol for Ruby, Java, and Python, but for my Python-3 project, it is missing. How do I add it? 

<img width="956" alt="image" src="https://github.com/community/community/assets/12854978/2eb136d4-817a-40b5-b7ef-a73c8e381c51">

**Comment by Aadil-H-W** [comment]
GitHub automatically detects the primary programming language of a repository based on the files within it, but sometimes it might not detect it correctly or at all. Here's how you can manually set the programming language for your repository:

1.Navigate to Your Repository: Go to the GitHub repository page for your project.

2.Click on "Settings": At the top-right corner of the repository page, you'll find a button labeled "Settings". Click on it to access the settings for your repository.

3.Go to the "Options" Section: In the settings sidebar on the left side of the page, scroll down and click on the "Options" section.
Choose a Language: In the "Options" section, you'll find a field labeled "Primary programming language". Click on the dropdown menu next to it.

4.Select the Language: From the dropdown menu, select the programming language that best represents your project. If the language you want isn't listed, you can manually type it in.

5.Save Changes: Once you've selected the appropriate language, scroll to the bottom of the page and click the "Save changes" button to apply the changes.

**Comment by rkhattab** [comment]
Thank you, I am just not able to see the options section after I click settings, this is what I see: 

<img width="1264" alt="image" src="https://github.com/community/community/assets/12854978/735795db-1efe-4af1-a0b0-44a9316228ec">

Could you clarify that step 3?
> 
> 3.Go to the "Options" Section: In the settings sidebar on the left side of the page, scroll down and click on the "Options" section. Choose a Language: In the "Options" section, you'll find a field labeled "Primary programming language". Click on the dropdown menu next to it.
>

**Comment by rkhattab** [comment, marked as answer]
Hey everyone I figured out the issue. It was kind of a stupid mistake, but I did not add the ".py" at the end of my file names. That is why Github was unable to detect it. Also if you create a repo, and there are no files in it yet, then of course GitHub won't be able to autodetect it.
